### PR TITLE
14.0 mig crm phonecall

### PR DESCRIPTION
**Pull request by redapureskill**
Migrating crm_phonecall to from V13 to **V14**.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -48,6 +48,7 @@ repos:
           - "@prettier/plugin-xml@0.12.0"
         args:
           - --plugin=@prettier/plugin-xml
+        files: \.xml$
```
Comment (rousseldenis) [inline .pre-commit-config.yaml:51]: Why changing this?

File: crm_phonecall/__manifest__.py [hunk 2]
```diff
@@ -1,11 +1,10 @@
-# Copyright 2017 Tecnativa - Vicent Cubells
```
Comment (rousseldenis) [inline crm_phonecall/__manifest__.py]: You cannot remove authorship

File: crm_phonecall/__manifest__.py [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
-# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+# For more information contact me on mireda2012@gmail.com
 
 {
     "name": "CRM Phone Calls",
-    "version": "13.0.1.2.2",
+    "version": "14.0.1.0.0",
     "category": "Customer Relationship Management",
-    "author": "Odoo S.A., Tecnativa, Odoo Community Association (OCA)",
+    "author": "Odoo Community Association (OCA)",
```
Comment (rousseldenis) [inline crm_phonecall/__manifest__.py]: Same here

File: setup/crm_phonecall/odoo/addons/crm_phonecall/crm_phonecall [hunk 4]
```diff
@@ -0,0 +1 @@
+../../../../crm_phonecall
```
Comment (rousseldenis) [inline setup/crm_phonecall/odoo/addons/crm_phonecall/crm_phonecall:1]: Mmmh, I think your link is not good (see Travis error). Did you use setuptools-odoo-make-default ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Hi @redapureskill, thank you for your contribution. Could you adapt your code ?

File: crm_phonecall/__manifest__.py (hunk 3 above)
Comment (redapureskill) [reply crm_phonecall/__manifest__.py]: git was showing problem with "no Odoo Community Association (OCA)" author found in manifest.
i'll fix it

File: crm_phonecall/__manifest__.py (hunk 2 above)
Comment (redapureskill) [reply crm_phonecall/__manifest__.py]: sure my bad

File: setup/crm_phonecall/odoo/addons/crm_phonecall/crm_phonecall (hunk 4 above)
Comment (redapureskill) [reply setup/crm_phonecall/odoo/addons/crm_phonecall/crm_phonecall:1]: i was getting this erreur when i do it : 
OSError: [WinError 1314] Le client ne dispose pas d'un privilège nécessaire: '..\\..\\..\\..\\crm_phonecall' -> '.\\setup\\crm_phonecall\\odoo\\addons\\crm_phonecall'
i tried with administrator same thing.

**Comment by pedrobaeza** [comment]
Are you developing in Windows? If so, that's the problem I'm afraid.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (redapureskill) [reply .pre-commit-config.yaml:51]: was giving me error on prettier and xml plugin in pre-commit, when i added it, it worked 👍

**Review by redapureskill — COMMENTED** [review]
Windows and docker

**Comment by pedrobaeza** [comment]
I think the flow is not well supported by Windows, as there's no symbolic link option there.

**Comment by rousseldenis** [comment]
@redapureskill The prettier problem should almost be changed in a separate commit (and best in another PR). As prettier for xml is quite new in OCA, I suggest you to point out what's going wrong (with error message) and take a look there https://github.com/OCA/oca-addons-repo-template/issues/34 if it is not the problem you have.

**Comment by redapureskill** [comment]
Generate default setup.py for an addons directory........................Failed
- hook id: setuptools-odoo-make-default
- exit code: 1

Traceback (most recent call last):
  File "C:\Users\y.yahla\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\y.yahla\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\y.yahla\.cache\pre-commit\repo_uveda\py_env-python3\Scripts\setuptools-odoo-make-default.EXE\__main__.py", line 7, in <module>
  File "c:\users\y.yahla\.cache\pre-commit\repo_uveda\py_env-python3\lib\site-packages\setuptools_odoo\make_default_setup.py", line 397, in main
    make_default_setup_addons_dir(
  File "c:\users\y.yahla\.cache\pre-commit\repo_uveda\py_env-python3\lib\site-packages\setuptools_odoo\make_default_setup.py", line 151, in make_default_setup_addons_dir
    make_default_setup_addon(
  File "c:\users\y.yahla\.cache\pre-commit\repo_uveda\py_env-python3\lib\site-packages\setuptools_odoo\make_default_setup.py", line 121, in make_default_setup_addon
    os.symlink(os.path.relpath(addon_dir, odoo_addons_path), link_path)
OSError: [WinError 1314] Le client ne dispose pas d'un privilège nécessaire: '..\\..\\..\\..\\crm_phonecall' -> '.\\setup\\crm_phonecall\\odoo\\addons\\crm_phonecall'

**Comment by redapureskill** [comment]
> Generate default setup.py for an addons directory........................Failed
> 
> * hook id: setuptools-odoo-make-default
> * exit code: 1
> 
> Traceback (most recent call last):
> File "C:\Users\y.yahla\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
> return _run_code(code, main_globals, None,
> File "C:\Users\y.yahla\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code exec(code, run_globals) File "C:\Users\y.yahla.cache\pre-commit\repo_uveda\py_env-python3\Scripts\setuptools-odoo-make-default.EXE__main__.py", line 7, in
> File "c:\users\y.yahla.cache\pre-commit\repo_uveda\py_env-python3\lib\site-packages\setuptools_odoo\make_default_setup.py", line 397, in main
> make_default_setup_addons_dir(
> File "c:\users\y.yahla.cache\pre-commit\repo_uveda\py_env-python3\lib\site-packages\setuptools_odoo\make_default_setup.py", line 151, in make_default_setup_addons_dir
> make_default_setup_addon(
> File "c:\users\y.yahla.cache\pre-commit\repo_uveda\py_env-python3\lib\site-packages\setuptools_odoo\make_default_setup.py", line 121, in make_default_setup_addon
> os.symlink(os.path.relpath(addon_dir, odoo_addons_path), link_path)
> OSError: [WinError 1314] Le client ne dispose pas d'un privilège nécessaire: '..\..\..\..\crm_phonecall' -> '.\setup\crm_phonecall\odoo\addons\crm_phonecall'

this is what's stopping me rn :'(

**Comment by pedrobaeza** [comment]
It seems you are forced to develop in Linux.

**Comment by redapureskill** [comment]
> It seems you are forced to develop in Linux.

i wonder if there's a docker image with git+python ?

**Comment by pedrobaeza** [comment]
Well, I think a base Ubuntu Docker image should be enough for having the proper stack, but I recommend you that if you are going to be serious in this Odoo world, to fully switch to Linux in your development environment.

**Comment by redapureskill** [comment]
i am, and i'll be switching to linux , 
thank you pedro

**Comment by redapureskill** [comment]
on linux, i'm getting errors on shebangs in pre-commit 👍 
Check that executables have shebangs.....................................Failed
- hook id: check-executables-have-shebangs
- exit code: 1

crm_phonecall/report/crm_phonecall_report_view.xml: marked executable but has no (or invalid) shebang!
  If it isn't supposed to be executable, try: `chmod -x crm_phonecall/report/crm_phonecall_report_view.xml`
  If it is supposed to be executable, double-check its shebang.
...

and 

pylint with mandatory checks.............................................Failed
- hook id: pylint
- exit code: 4

************* Module crm_phonecall.report.__init__
crm_phonecall/report/__init__.py:1: [W8201(incoherent-interpreter-exec-perm), ] You have a python file with execution permissions but you don't have an interpreter magic comment, or a magic comment but no execution permission. If you really needs a execution permission then add a magic comment ( https://en.wikipedia.org/wiki/Shebang_(Unix) ). If you don't needs a execution permission then remove it with: chmod -x __init__.py
...

**Comment by pedrobaeza** [comment]
Are you sharing the files through NTFS or similar (or copied from them)? The +x attribute is not allowed in Windows like filesystems, and then Linux put to all of them this flag by default. You should clone the repo directly on Linux or change these attributes.

**Comment by redapureskill** [comment]
i tried changing the attributes using chmod -x but still same errors.

**Comment by pedrobaeza** [comment]
As said, it seems you are under a Window filesystem.

**Comment by redapureskill** [comment]
Should be ok in #364

**Pull request closed without merging**